app/components/channel_link/channel_link.js

```diff
@@ -39,12 +50,35 @@ export default class ChannelLink extends React.PureComponent {
         return null;
     }
 
-    handlePress = () => {
-        this.props.actions.setChannelDisplayName(this.state.channel.display_name);
-        this.props.actions.handleSelectChannel(this.state.channel.id);
+    handlePress = async () => {
+        const {channel: channelState} = this.state;
+        let channel;
+        if (!channelState.id && channelState.display_name) {
+            const {actions, currentTeamId, currentUserId} = this.props;
+            const result = await actions.joinChannel(currentUserId, currentTeamId, null, channelState.display_name);
```

Comment:
This needs to use the channel's name, not the display name. We might need to modify `getChannelFromChannelName` to have it also return the matched name instead of just the channel

Reply:
Oh you're right.  Now updated.